src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2823,6 +2831,18 @@ ErrorIfContainsUnsupportedSubquery(MultiNode *logicalPlanNode)
 	/* currently in the planner we only allow one subquery in from-clause*/
 	Assert(list_length(subqueryMultiTableList) == 1);
 
+	allRelationsJoinedOnPartitionKey =
+		AllRelationsJoinedOnPartitionKey(plannerRestrictionContext);
```

Comment:
I still think that this is a misleading name in the presence of reference tables.